src/Widgets/EntryPopover.vala

```diff
@@ -0,0 +1,134 @@
+    public signal void value_changed (T value);
+    public signal string? value_format (T value);
+
+    private static Gtk.CssProvider style_provider;
+    private Gtk.MenuButton popover_button;
+
+    class construct {
+        set_css_name ("entry-popover");
+    }
+
+    static construct {
+        style_provider = new Gtk.CssProvider ();
+        style_provider.load_from_resource ("io/elementary/tasks/EntryPopover.css");
+    }
+
+    construct {
+        events |= Gdk.EventMask.ENTER_NOTIFY_MASK
+            | Gdk.EventMask.LEAVE_NOTIFY_MASK;
+
+        popover = new Gtk.Popover (popover_button);
+
+        popover_button = new Gtk.MenuButton () {
+            always_show_image = true,
+            label = (placeholder != null && placeholder.length > 0 ? placeholder : _("Set Value")),
+            popover = popover
+        };
+        popover_button.get_style_context ().add_class (Gtk.STYLE_CLASS_FLAT);
+
+        var delete_button = new Gtk.Button.from_icon_name ("process-stop-symbolic", Gtk.IconSize.BUTTON) {
+            tooltip_text = _("Remove")
+        };
+        delete_button.get_style_context ().add_class (Gtk.STYLE_CLASS_FLAT);
+
+        var delete_button_revealer = new Gtk.Revealer () {
+            transition_type = Gtk.RevealerTransitionType.SLIDE_RIGHT,
+            reveal_child = false
+        };
+        delete_button_revealer.add (delete_button);
+
+        var button_box = new Gtk.Grid ();
+        button_box.add (popover_button);
+        button_box.add (delete_button_revealer);
+        button_box.get_style_context ().add_provider (style_provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
+
+        add (button_box);
+
+        bind_property ("image", popover_button, "image");
+
+        delete_button.clicked.connect (() => {
+            var value_has_changed = value != null;
+            value = null;
+            if (value_has_changed) {
+                value_changed (value);
+            }
+        });
+
+        popover_button.clicked.connect (() => {
+            if (delete_button_revealer.reveal_child) {
+                delete_button_revealer.reveal_child = false;
+            }
+        });
+
+        notify["placeholder"].connect (() => {
+            if (value_format (value) == null) {
+                popover_button.label = (placeholder != null && placeholder.length > 0 ? placeholder : _("Set Value"));
+            }
+        });
+
+        notify["value"].connect (() => {
+            var value_formatted = value_format (value);
+            if (value_formatted == null) {
+                popover_button.label = (placeholder != null && placeholder.length > 0 ? placeholder : _("Set Value"));
+
+                if (delete_button_revealer.reveal_child) {
+                    Timeout.add (150, () => {
+                        delete_button_revealer.reveal_child = false;
+                        return GLib.Source.REMOVE;
+                    });
+                }
```

Comment:
Can you explain what this is for?

Reply:
The Timeout only serves a cosmetic purpose: If the user clicks on the delete button, the placeholder gets displayed immediately. Therefore it felt nicer to me to wait a bit till this change is visually done - and only after this, start the revealer's hiding transition.

Reply:
gave it another try without the Timeout. Feels no longer weird, therefore removed the Timeout.